main.tf

```diff
@@ -359,6 +359,10 @@ resource "aws_s3_bucket" "aws_logs" {
   policy        = data.aws_iam_policy_document.main.json
   force_destroy = var.force_destroy
 
+  versioning {
```

Comment:
Does this need to be wrapped in a dynamic block?

If you set `enabled = false` what does that do?

According to resources docs, you can only suspend versioning after the fact so if you try to enable => disable what does it do?

Reply:
I don't think it needs to be in a dynamic block -- if versioning is false, then it's set to false, if it's true, it's set to true. It doesn't appear that it cares if `noncurrent_version_expiration` exists if versioning is set to false (since all tests pass).

As for the "you can't turn off versioning, only suspend it," that's a good question. I've never tried turning off versioning via Terraform...not sure how we would do that in an automated test, but I can try messing with it for funsies manually.

Reply:
So if you set it to false after it was previously set to true, Terraform will accept it without complaint. Checking in the console, the bucket is then in "suspended" mode, which seems like the behavior you want.